src/queries.js

```diff
@@ -8,18 +8,18 @@ import { query, querySync } from 'avenger';
 
 const log = debug('react-avenger:queries');
 
-const mapQueriesToState = ({ data }) =>
-  Object.keys(data).reduce(
-    (ac, k) => ({
+const mapQueriesToState = ({ data }, prevState) =>
+  Object.keys(data).reduce((ac, k) => {
+    const value = 'data' in data[k] ? data[k].data : (prevState[k] && prevState[k].data) || void 0;
```

Comment:
Ok, I'm finally convinced this this works but I spent 2 minutes reading it so I guess we could improve it in readability :D

the confusing part is the last one where you're relying on the `||` so that when `prevState[k].data` is `undefined` you actually return `void 0` instead of `undefined`